File: SensorHook.java
package com.sensorsoff;

import de.robv.android.xposed.IXposedHookLoadPackage;
import de.robv.android.xposed.callbacks.XC_LoadPackage;
import de.robv.android.xposed.XposedHelpers;
import de.robv.android.xposed.XC_MethodHook;

public class SensorHook implements IXposedHookLoadPackage {
    @Override
    public void handleLoadPackage(final XC_LoadPackage.LoadPackageParam lpparam) throws Throwable {
        if (lpparam.packageName.equals("com.android.systemui")) {
            try {
                Class<?> sensorManagerClass = XposedHelpers.findClass(
                        "android.hardware.SensorManager",
                        lpparam.classLoader
                );

                XposedHelpers.findAndHookMethod(
                        sensorManagerClass,
                        "registerListener",
                        int.class,
                        boolean.class,
                        new XC_MethodHook() {
                            @Override
                            protected void beforeHookedMethod(MethodHookParam param) {
                                int sensorType = (int) param.args[0];
                                if (sensorType == 9 || sensorType == 10) { // Gyroscope/Orientation
                                    param.args[1] = true; // Keep mic enabled
                                }
                            }
                        }
                );
            } catch (Exception e) {
                XposedHelpers.log("SensorHook Error: " + e.getMessage());
            }
        }
    }
}

File: SensorHook.kt
package com.yourpackage.sensorsmod

import de.robv.android.xposed.IXposedHookLoadPackage
import de.robv.android.xposed.callbacks.XC_LoadPackage
import de.robv.android.xposed.XposedHelpers
import de.robv.android.xposed.XC_MethodHook

class SensorHook : IXposedHookLoadPackage {
    override fun handleLoadPackage(lpparam: XC_LoadPackage.LoadPackageParam) {
        if (lpparam.packageName == "com.android.systemui") {
            try {
                val sensorManagerClass = XposedHelpers.findClass(
                    "android.hardware.SensorManager", 
                    lpparam.classLoader
                )

                XposedHelpers.findAndHookMethod(
                    sensorManagerClass, "setSensorEnabled",
                    Int::class.java, Boolean::class.java, object : XC_MethodHook() {
                        override fun beforeHookedMethod(param: MethodHookParam) {
                            val sensorType = param.args[0] as Int
                            if (sensorType == SENSOR_MICROPHONE) {
                                param.args[1] = true // Always enable mic
                            }
                        }
                    }
                )
            } catch (e: Exception) {
                XposedHelpers.log("SensorHook: Error - ${e.message}")
            }
        }
    }

    companion object {
        private const val SENSOR_MICROPHONE = 1 // Replace with correct sensor ID if needed
    }
}
